src/hooks/useDetails.tsx

```diff
@@ -1,13 +1,23 @@
 import {useCallback, useEffect, useState, useRef} from 'react'
 
-function useDetails({ref, closeOnOutsideClick, defaultOpen, onClickOutside} = {}) {
+type useDetailsParameters = {
```

Comment:
I think it's a convention to capitalize the first letter of types:

```suggestion
type UseDetailsParameters = {
```